workflow/setup_xml.py

```diff
@@ -48,6 +48,33 @@ def check_expdir(cmd_expdir, cfg_expdir):
         print(f'  input arg: --expdir = {cmd_expdir}')
         raise ValueError('Abort!')
 
+def check_dir_writable(dirPath):
+    if os.path.isdir(dirPath):
+        if os.access(dirPath, os.W_OK):
+            print(f'{dirPath} is writable!')
+            return True
+        else:
+            raise PermissionError(f"{dirPath} is not writable. Please change its permission mode as writable.")
+            return False
```

Comment:
I'd suggest printing an informative message here and returning `False` instead of raising an error.  Let the calling function handle the error.

```suggestion
            print(f"{dirPath} is unwritable!")
            return False
```